server/controller_client_test.go

```diff
@@ -110,3 +113,11 @@ func TestAgentControl(t *testing.T) {
 	assert.True(t, gotHealthy, "healthy state")
 	assert.True(t, gotStopped, "stopped state")
 }
+
+func MustNewStruct(contents map[string]interface{}) *structpb.Struct {
+	result, err := structpb.NewStruct(contents)
+	if err != nil {
+		panic(fmt.Errorf("failed to create test struct for contents: %w", err))
```

Comment:
Could you pass in a `testing.T` instance instead and call `t.Fatal` instead of panicing?